sync/task_pool_test.go

```diff
@@ -252,21 +404,33 @@ func BenchmarkTaskPool_RandomTask(b *testing.B) {
 }
 
 /*
-
+goos: darwin
+goarch: amd64
 pkg: github.com/dubbogo/gost/sync
-BenchmarkTaskPool_CountTask/AddTask-8         	 2872177	       380 ns/op	       0 B/op	       0 allocs/op
-BenchmarkTaskPool_CountTask/AddTaskAlways-8   	 2769730	       455 ns/op	       1 B/op	       0 allocs/op
-BenchmarkTaskPool_CountTask/AddTaskBalance-8  	 4630167	       248 ns/op	       0 B/op	       0 allocs/op
-BenchmarkTaskPool_CPUTask/fib-8               	   73975	     16524 ns/op	       0 B/op	       0 allocs/op
-BenchmarkTaskPool_CPUTask/AddTask-8           	   72525	     18160 ns/op	       0 B/op	       0 allocs/op
-BenchmarkTaskPool_CPUTask/AddTaskAlways-8     	  606813	     16464 ns/op	      40 B/op	       0 allocs/op
-BenchmarkTaskPool_CPUTask/AddTaskBalance-8    	  137926	     17646 ns/op	       0 B/op	       0 allocs/op
-BenchmarkTaskPool_IOTask/AddTask-8            	   10000	    108520 ns/op	       0 B/op	       0 allocs/op
-BenchmarkTaskPool_IOTask/AddTaskAlways-8      	 1000000	      1236 ns/op	      95 B/op	       1 allocs/op
-BenchmarkTaskPool_IOTask/AddTaskBalance-8     	 1518144	       673 ns/op	      63 B/op	       0 allocs/op
-BenchmarkTaskPool_RandomTask/AddTask-8        	  497055	      2517 ns/op	       6 B/op	       0 allocs/op
-BenchmarkTaskPool_RandomTask/AddTaskAlways-8  	 2511391	       415 ns/op	      21 B/op	       0 allocs/op
-BenchmarkTaskPool_RandomTask/AddTaskBalance-8 	 1381711	       868 ns/op	      17 B/op	       0 allocs/op
+BenchmarkTaskPoolSimple_CountTask/AddTask-8              1693192               700 ns/op               0 B/op          0 allocs/op
```

Comment:
I think there should be a document to tell user when to use the TaskPool , and when to use the SimpleTaskTool?

Reply:
agree with u

Reply:
The data as follows, run 100000 tasks and with 1000 size of taskpool.
```golang
// IOFunction
func newIOTask() (func(), *int64) {
	var cnt int64
	return func() {
		time.Sleep(700 * time.Microsecond)
	}, &cnt
}

=== RUN   TestWithoutPool
--- PASS: TestWithoutPool (0.04s)
    task_pool_test.go:384: Before Alloc = 0 MiB
    task_pool_test.go:385:      TotalAlloc = 0 MiB
    task_pool_test.go:386:      Sys = 66 MiB
    task_pool_test.go:387:      NumGC = 0
    task_pool_test.go:416: TestWithoutPool took 3460 goroutines
    task_pool_test.go:384: After Alloc = 2 MiB
    task_pool_test.go:385:      TotalAlloc = 8 MiB
    task_pool_test.go:386:      Sys = 69 MiB
    task_pool_test.go:387:      NumGC = 3
    task_pool_test.go:393: 
                 TestWithoutPool took 36.100024ms
PASS

=== RUN   TestWithSimplePool
--- PASS: TestWithSimplePool (0.13s)
    task_pool_test.go:385: Before Alloc = 0 MiB
    task_pool_test.go:386:      TotalAlloc = 0 MiB
    task_pool_test.go:387:      Sys = 68 MiB
    task_pool_test.go:388:      NumGC = 0
    task_pool_test.go:445: TestWithSimplePool took 1000 goroutines
    task_pool_test.go:385: After Alloc = 0 MiB
    task_pool_test.go:386:      TotalAlloc = 0 MiB
    task_pool_test.go:387:      Sys = 68 MiB
    task_pool_test.go:388:      NumGC = 0
    task_pool_test.go:394: 
                 TestWithSimplePool took 129.873271ms
PASS

=== RUN   TestWithSimplePoolUseAlways
--- PASS: TestWithSimplePoolUseAlways (0.06s)
    task_pool_test.go:385: Before Alloc = 0 MiB
    task_pool_test.go:386:      TotalAlloc = 0 MiB
    task_pool_test.go:387:      Sys = 66 MiB
    task_pool_test.go:388:      NumGC = 0
    task_pool_test.go:431: TestWithSimplePool took 2048 goroutines
    task_pool_test.go:385: After Alloc = 2 MiB
    task_pool_test.go:386:      TotalAlloc = 4 MiB
    task_pool_test.go:387:      Sys = 69 MiB
    task_pool_test.go:388:      NumGC = 1
    task_pool_test.go:394: 
                 TestWithSimplePool took 60.207644ms
PASS

=== RUN   TestWithPool
--- PASS: TestWithPool (0.14s)
    task_pool_test.go:384: Before Alloc = 0 MiB
    task_pool_test.go:385:      TotalAlloc = 0 MiB
    task_pool_test.go:386:      Sys = 66 MiB
    task_pool_test.go:387:      NumGC = 0
    task_pool_test.go:462: TestWithPool took 1000 goroutines
    task_pool_test.go:384: After Alloc = 0 MiB
    task_pool_test.go:385:      TotalAlloc = 0 MiB
    task_pool_test.go:386:      Sys = 68 MiB
    task_pool_test.go:387:      NumGC = 0
    task_pool_test.go:393: 
                 TestWithPool took 135.168986ms
PASS

=== RUN   TestWithPoolUseAlways
--- PASS: TestWithPoolUseAlways (0.07s)
    task_pool_test.go:384: Before Alloc = 0 MiB
    task_pool_test.go:385:      TotalAlloc = 0 MiB
    task_pool_test.go:386:      Sys = 66 MiB
    task_pool_test.go:387:      NumGC = 0
    task_pool_test.go:480: TestWithPoolUseAlways took 3280 goroutines
    task_pool_test.go:384: After Alloc = 2 MiB
    task_pool_test.go:385:      TotalAlloc = 5 MiB
    task_pool_test.go:386:      Sys = 69 MiB
    task_pool_test.go:387:      NumGC = 2
    task_pool_test.go:393: 
                 TestWithPoolUseAlways took 65.361076ms
PASS
```